### PR TITLE
expression: implement vectorized evaluation for builtinCastRealAsTimeSig

### PR DESCRIPTION
**Pull request by ekalinin**
### What problem does this PR solve? 

This PR implements vectorized `builtinCastRealAsTimeSig`. See #12101

### What is changed and how it works?

```
➜ make vectorized-bench VB_FILE=Cast VB_FUNC=builtinCastRealAsTimeSig
cd ./expression && \
	go test -v -benchmem \
		-bench=BenchmarkVectorizedBuiltinCastFunc \
		-run=BenchmarkVectorizedBuiltinCastFunc \
		-args "builtinCastRealAsTimeSig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsTimeSig-VecBuiltinFunc-12         	     514	   2167453 ns/op	  521584 B/op	   12225 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsTimeSig-NonVecBuiltinFunc-12      	     386	   3054068 ns/op	  521584 B/op	   12225 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.900s
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by ekalinin** [comment]
> @ekalinin , Thanks for your contribution. Almost LGTM. Could you please add some test cases?

I believe there're some tests here:
- https://github.com/pingcap/tidb/blob/06342ddbe28851ef7ba19ffc0c7e20b0c3cb49c3/expression/builtin_cast_test.go#L757

**Comment by ekalinin** [comment]
> @ekalinin , Thanks for your contribution. Almost LGTM. Could you please add some test cases?

@Reminiscent sorry, forgot to add `expression/builtin_cast_vec_test.go` into first commit.

Fixed.

File: expression/builtin_cast_vec_test.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ var vecBuiltinCastCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETJson}},
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETString}, geners: []dataGenerator{&jsonStringGener{}}},
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETDecimal}},
+		{retEvalType: types.ETDatetime, childrenTypes: []types.EvalType{types.ETReal}},
```
Comment (SunRunAway) [inline expression/builtin_cast_vec_test.go]: Please add cases for `mysql.TypeDate`, we can use [`childrenFieldTypes`](https://github.com/pingcap/tidb/blob/2d28a2703f50c99642bdaec7e47901da79187536/expression/bench_test.go#L472) here.

Comment (ekalinin) [reply expression/builtin_cast_vec_test.go]: Could you provide more details, please?
if I understand correctly, `childrenFieldTypes` is for input datatype only?
How can we change output type?

**Pull request merged**